### ISSUE TITLE
e57 warnings on osx-arm64

### ISSUE THREAD

**Issue by hobu** [PDAL/PDAL#3579]
```
/Users/hobu/miniforge3/envs/conda-build/conda-bld/pdal_1635971947921/work/plugins/e57/libE57Format/src/SourceDestBufferImpl.cpp:366:43: warning: implicit conversion from 'const int64_t' (aka 'const long long') to 'double' changes value from 9223372036854775807 to 9223372036854775808 [-Wimplicit-const-int-float-conversion]
    if (doubleRawValue < E57_INT64_MIN || E57_INT64_MAX < doubleRawValue) {
                                          ^~~~~~~~~~~~~ ~
/Users/hobu/miniforge3/envs/conda-build/conda-bld/pdal_1635971947921/work/plugins/e57/libE57Format/src/SourceDestBufferImpl.cpp:150:41: warning: implicit conversion from 'const int32_t' (aka 'const int') to 'float' changes value from 2147483647 to 2147483648 [-Wimplicit-const-int-float-conversion]
         if (inValue < E57_INT32_MIN || E57_INT32_MAX < inValue)
                                        ^~~~~~~~~~~~~ ~
/Users/hobu/miniforge3/envs/conda-build/conda-bld/pdal_1635971947921/work/plugins/e57/libE57Format/src/SourceDestBufferImpl.cpp:697:4: note: in instantiation of function template specialization 'e57::SourceDestBufferImpl::_setNextReal<float>' requested here
   _setNextReal( value );
   ^
/Users/hobu/miniforge3/envs/conda-build/conda-bld/pdal_1635971947921/work/plugins/e57/libE57Format/src/SourceDestBufferImpl.cpp:157:42: warning: implicit conversion from 'const uint32_t' (aka 'const unsigned int') to 'float' changes value from 4294967295 to 4294967296 [-Wimplicit-const-int-float-conversion]
         if (inValue < E57_UINT32_MIN || E57_UINT32_MAX < inValue)
                                         ^~~~~~~~~~~~~~ ~
/Users/hobu/miniforge3/envs/conda-build/conda-bld/pdal_1635971947921/work/plugins/e57/libE57Format/src/SourceDestBufferImpl.cpp:164:41: warning: implicit conversion from 'const int64_t' (aka 'const long long') to 'float' changes value from 9223372036854775807 to 9223372036854775808 [-Wimplicit-const-int-float-conversion]
         if (inValue < E57_INT64_MIN || E57_INT64_MAX < inValue)
                                        ^~~~~~~~~~~~~ ~
[ 10%] Building CXX object plugins/e57/libE57Format/CMakeFiles/E57Format.dir/src/E57Exception.cpp.o
/Users/hobu/miniforge3/envs/conda-build/conda-bld/pdal_1635971947921/work/plugins/e57/libE57Format/src/SourceDestBufferImpl.cpp:164:41: warning: implicit conversion from 'const int64_t' (aka 'const long long') to 'double' changes value from 9223372036854775807 to 9223372036854775808 [-Wimplicit-const-int-float-conversion]
         if (inValue < E57_INT64_MIN || E57_INT64_MAX < inValue)
                                        ^~~~~~~~~~~~~ ~
/Users/hobu/miniforge3/envs/conda-build/conda-bld/pdal_1635971947921/work/plugins/e57/libE57Format/src/SourceDestBufferImpl.cpp:702:5: note: in instantiation of function template specialization 'e57::SourceDestBufferImpl::_setNextReal<double>' requested here
    _setNextReal( value );
    ^
```

**Comment by abellgithub** [comment]
I believe these warnings can be eliminated by casting the constants to the proper type (double/float) before the comparison.

Because the double created by the cast is one more than the integral value, any value less than that, when converted to an integer, should be no more than the E57_INT<x>_MAX value.

**Comment by hobu** [comment]
Addressed in #3589
